tests/chocolatey-tests/choco-install.Tests.ps1

```diff
@@ -0,0 +1,1378 @@
+            "FileNotOverwritten" | Add-Content -Path "$env:ChocolateyInstall\lib\$PackageUnderTest\tools\chocolateyInstall.ps1"
+
+            $Output = Invoke-Choco install $PackageUnderTest --version 1.0.0 --force --confirm
+        }
+
+        It "Exits with Success (0)" {
+            $Output.ExitCode | Should -Be 0
+        }
+
+        It "Should install the package in the lib directory" {
+            "$env:ChocolateyInstall\lib\$PackageUnderTest" | Should -Exist
+        }
+
+        It "Should still have the expected version installed" {
+            "$env:ChocolateyInstall\lib\$PackageUnderTest\$PackageUnderTest.nuspec" | Should -Exist
+            [xml]$XML = Get-Content "$env:ChocolateyInstall\lib\$PackageUnderTest\$PackageUnderTest.nuspec"
+            $XML.package.metadata.version | Should -Be "1.0.0"
+        }
+
+        It "Should remove and readd the package files in the lib directory" {
+            "$env:ChocolateyInstall\lib\$PackageUnderTest\tools\chocolateyInstall.ps1" | Should -Not -FileContentMatch "FileNotOverwritten"
+        }
+
+        It "Deletes the rollback files" {
+            "$env:ChocolateyInstall\lib-bkp\$PackageUnderTest" | Should -Not -Exist
+        }
+
+        It "Outputs a message indicating that it installed the package successfully" {
+            $Output.Lines | Should -Contain "Chocolatey installed 1/1 packages."
+        }
+    }
+
+    Context "Force Installing a Package that is already installed (with a read/delete locked file)" {
+        BeforeAll {
+            Restore-ChocolateyInstallSnapshot
+
+            $PackageUnderTest = "installpackage"
+
+            Invoke-Choco install $PackageUnderTest --confirm
+
+            $LockedFile = [System.IO.File]::Open(
+                "$env:ChocolateyInstall\lib\$PackageUnderTest\tools\chocolateyInstall.ps1",
+                "OpenOrCreate",
+                "ReadWrite",
+                "Read, Delete"
+            )
+
+            $Output = Invoke-Choco install installpackage --force --confirm
+        }
+
+        AfterAll {
+            $LockedFile.Close()
+        }
+
+        It "Exits with Success (0)" {
+            $Output.ExitCode | Should -Be 0
+        }
+
+        It "Has successfully retained an install of the original package" {
+            "$env:ChocolateyInstall\lib\$PackageUnderTest\"
+        }
+
+        It "Has successfully retained the original version" {
+            "$env:ChocolateyInstall\lib\$PackageUnderTest\$PackageUnderTest.nuspec" | Should -Exist
+            [xml]$XML = Get-Content "$env:ChocolateyInstall\lib\$PackageUnderTest\$PackageUnderTest.nuspec"
+            $XML.package.metadata.version | Should -Be "1.0.0"
+        }
+
+        # This behaviour was fixed in 0.10.16
+        It "Should not have been able to delete the rollback" -Skip:(-not (Test-ChocolateyVersionEqualOrHigherThan "0.10.16-beta")) {
+            "$env:ChocolateyInstall\lib-bkp\$PackageUnderTest" | Should -Exist
+        }

```

Comment:
Maybe we should remove this test if it's since been fixed and replace it with a test that it _should_ have been able to delete it?

Reply:
That's the thing, if I'm reading the Skip and the comment correctly, this is indicating that the test is to verify that it was fixed.

Reply:
Ahh okay, I think I see.